### PR TITLE
Add gdb pretty printer for cv::Mat

### PR DESCRIPTION
**Pull request by rogday**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [X] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/gdb/mat_pretty_printer.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        dataptr = int(m['data'])
+        length = (int(m['dataend']) - dataptr) // elsize
+        start = (int(m['datastart']) - dataptr) // elsize
+
+        ptr = m['data']
+
+        if dtype != np.float16:
+            ctype = gdb.lookup_type(ctype)
+            ptr = ptr.cast(ctype.array(length - 1).pointer()).dereference()
+            self.mat = np.array([ptr[i] for i in range(length)], dtype=dtype)
```
Comment (sergregory) [inline samples/gdb/mat_pretty_printer.py:186]: probably, we iterate over all items in `ptr` here? Are there any scenarios when length != len(ptr) ?

Comment (rogday) [reply samples/gdb/mat_pretty_printer.py:186]: Yes, we do iterate over all items in ptr. But ptr is not iterable.

File: samples/gdb/mat_pretty_printer.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        else:
+            u16 = gdb.lookup_type('uint16_t')
+            ptr = ptr.cast(u16.array(length - 1).pointer()).dereference()
+            self.mat = np.array([ptr[i] for i in range(length)], dtype=np.uint16)
+            self.mat = self.mat.view(np.float16)
+
+        steps = np.asarray(list(map(int, to_list(m['step']['p'], range(size.dims())))), dtype=np.int64)
```
[the remark this answers is not in the document]
Comment (rogday) [reply samples/gdb/mat_pretty_printer.py]: Fixed.

File: samples/gdb/mat_pretty_printer.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        self.view = np.lib.stride_tricks.as_strided(self.mat[start:], shape=size.to_numpy(), strides=steps)
+
+    def __iter__(self):
+        return iter({'data': stri(self.view)}.items())
+
+
+class MatPrinter(object):
```
[the remark this answers is not in the document]
Comment (rogday) [reply samples/gdb/mat_pretty_printer.py]: Fixed.

File: samples/gdb/mat_pretty_printer.py (hunk 1 above)
Comment (sergregory) [reply samples/gdb/mat_pretty_printer.py:138]: Got it, thanks

**Review by sergregory — APPROVED** [review]

**Comment by sergregory** [comment]
LGTM
@asmorkalov , your turn

**Comment by asmorkalov** [comment]
@alalek Could you merge the PR?

**Pull request merged**